invenio_pidstore/provider.py

```diff
@@ -48,6 +48,14 @@ class PIDStatus:
     registered.
     """
 
+    @classmethod
+    def list_in_order(cls):
```

Comment:
:+1: for first suggestion

as for the second I'm not sure. If we are keeping the all-caps form I'd say it's better to keep it as non-translated "constants".

Perhaps making it explicit that it's a mapping of ``enum -> label``:
```
status_order = ((NEW, _('New'), (REGISTERED, _('Registered')))
```
but then It's a little bit of trespassing the UI domain.